ui/src/containers/AppVersionsTableContainer.js

```diff
@@ -9,20 +9,42 @@ import './TableContainer.css';
 import config from '../config/config';
 
 export class AppVersionsTableContainer extends React.Component {
+  state = {
+    canUpdateComponent: true
+  }
+
+  shouldComponentUpdate () {
+    return this.state.canUpdateComponent;
+  }
+
   handleDisableAppVersionChange = (_event, e) => {
     const id = e.target.id;
     const isDisabled = e.target.checked;
-    this.props.updateDisabledAppVersion(id, isDisabled);
+
+    this.setState({
+      canUpdateComponent: false
+    }, () => {
+      this.props.updateDisabledAppVersion(id, isDisabled);
```

Comment:
I think this is definitely the correct approach to handling this, just calling the reducer action from the onBlur event